### PR TITLE
Additional contexts & projects - cleanup of PR #1305 changes, closes #1302

### LINKED ISSUE #1302: todo.txt: Settings option for always-visible @contexts
Markor will list all the projects and contexts available in the `add projects` and `add contexts` dialogs respectively.

However, if I delete (or mark as done and archive) the last task with a context or project, it is no longer in the list and I have to type it again.

We should have a setting where a user can enter a (space separated) list of projects and contexts.

I.e for example

```
@home @pc @work +work +garage 
```
These will then always be available in the context

### PR DESCRIPTION
**Pull request by harshad1**
@gsantner Couple of simplifications to the code

### CODE REVIEW

File: app/src/main/java/net/gsantner/markor/ui/SearchOrCustomTextDialogCreator.java [hunk 1]
```diff
@@ -262,7 +263,7 @@ public static void showSttSortDialogue(Activity activity, final Callback.a2<Stri
     public static void showSttContextDialog(Activity activity, List<String> availableData, Callback.a1<String> callback) {
         SearchOrCustomTextDialog.DialogOptions dopt = new SearchOrCustomTextDialog.DialogOptions();
         baseConf(activity, dopt);
-        dopt.data = sortUniqNonEmpty(availableData);
+        dopt.data = new ArrayList<>(new TreeSet<>(availableData));
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/ui/SearchOrCustomTextDialogCreator.java:266]: Replace function with one-liner

File: app/src/main/java/net/gsantner/markor/util/AppSettings.java [hunk 2]
```diff
@@ -813,7 +813,7 @@ public boolean getSetWebViewFulldrawing(boolean... setValue) {
     }
 
     public String getTodotxtAdditionalContextsAndProjects() {
-        return getString(R.string.pref_key__todotxt__additional_projects_contexts, "+music +video @home @shop").replaceAll("\\s+", " ").trim();
+        return getString(R.string.pref_key__todotxt__additional_projects_contexts, "+music +video @home @shop");
```
Comment (harshad1) [inline app/src/main/java/net/gsantner/markor/util/AppSettings.java:816]: We don't require the replace and trim code. The additional string is parsed by the same logic which parses tasks so multiple and trailing spaces are ignored.

Comment (gsantner) [reply app/src/main/java/net/gsantner/markor/util/AppSettings.java:816]: edittext preference is multiline (no android: option). Task class is for single line of text?

Comment (harshad1) [reply app/src/main/java/net/gsantner/markor/util/AppSettings.java:816]: The task parser can handle multiple lines (i.e. text with `\n`). 

image

Works fine

**Review by gsantner — APPROVED** [review]

**Pull request merged**